### PR TITLE
eval_array and array functions

### PR DESCRIPTION
**Pull request by pjaspers**
- Add support for two new functions: `array` and `eval_array`

      eval_array('a', (1,2,3), 'b', (2,3,4), 'b - a')

      'a'				=> key 1
      '(1,2,3)'	=> array 1
      'b'				=> key 2
      '(2,3,4)		=> array 2 (needs to be same length as array 1)
      'b-a'			=> metaformula

      Will loop over the arrays and apply the formula to each index, so
      in this example would result in:

            (2-1, 3-2,4-3)
            (1,1,1)

      array(1,2,3,4,5)

      A noop function in this context, mainly added for api parity with
      dentaku, so arrays can be explicilty marked as arrays.

      ARRAY(1,2,3) => (1,2,3,4,5)

**NOTE** This depends on a new version of hesabu (which needs a new version of go-hesabu, so first I need to get the go version merged, then I can release a new version of the hesabu gem, and then these tests will pass

## Self proof reading checklist

- [x] Did I run Pronto and fixed all warnings
- [x] Make sure all naming and code will remain understandable in 6 month by someone new to the project or by you.
- [x] Did I test the right thing?
- [x] Did I test corner cases, non happy path (defensive testing)?
- [x] Check that I used the ad-hoc patterns and created files accordingly
- [x] Check code efficiency with record intensive project
- [x] Update documentation,readme accordingly

Thanks!

### CODE REVIEW

**Comment by mestachs** [comment]
with the profiling can you check if we shouldn't remove the log statements
https://github.com/BLSQ/go-hesabu/blob/master/hesabu/parser.go#L83

File: spec/lib/orbf/rules_engine/builders/calculator_factory_spec.rb [hunk 1]
```diff
@@ -75,6 +75,39 @@ def solve_score_table(value)
           expect(solution["my_var"]).to be_between(1, 4)
         end
       end
+
+      describe 'eval_array' do
+        it 'solves' do
+          solution = calculator.solve("my_var" => "sum(eval_array('a', ARRAY(1,3,4), 'b', ARRAY(2,4,5), 'b - a'))")
+          expect(solution["my_var"]).to eq(1+1+1)
```
Comment (mestachs) [inline spec/lib/orbf/rules_engine/builders/calculator_factory_spec.rb]: as we found issues with negative values, I would just tune this one with
```
calculator.solve("my_var" => "sum(eval_array('a', ARRAY(1.5,-3,4), 'b', ARRAY(2,-4,5), 'b - a'))")
expect(solution["my_var"]).to eq((2-1.5)+(-4+3)+(5-4))
```

Comment (mestachs) [reply spec/lib/orbf/rules_engine/builders/calculator_factory_spec.rb]: would add more complex expression (reuse existing function like if or max)

File: spec/lib/orbf/rules_engine/builders/calculator_factory_spec.rb [hunk 2]
```diff
@@ -75,6 +75,39 @@ def solve_score_table(value)
...
+        end
+
+        describe 'raises on invalid input' do
+          it 'not same length arrays' do
+            expect{
+              calculator.solve!("my_var" => "sum(eval_array('a', ARRAY(), 'b', ARRAY(1), 'b - a'))")
+            }.to raise_error(StandardError)
```
Comment (mestachs) [inline spec/lib/orbf/rules_engine/builders/calculator_factory_spec.rb]: I would assert on the message (standard error can be quiet common ;))
and as the message will show up to the user, would make sure they are meaningful
